docs/dev/gen-nb-src.py

```diff
@@ -0,0 +1,59 @@
+'''
+Transforms a .html notebook into its source .py/.scala/.r/,sql file.
+
+This script is used by the CircleCI job 'check-docs'. Before running this, configure
+your Databricks CLI profile.
+
+Example usage:
+  python3 docs/dev/gen-nb-src.py \
+    --html docs/source/_static/notebooks/etl/variant-data.html
+'''
+import click
+import subprocess
+import os
+import uuid
+
+NOTEBOOK_DIR = 'docs/source/_static/notebooks'
+SOURCE_DIR = 'docs/source/_static/zzz_GENERATED_NOTEBOOK_SOURCE'
+SOURCE_EXTS = ['scala', 'py', 'r', 'sql']
+
+
+@click.command()
+@click.option('--html', required=True, help='Path of the HTML notebook.')
+@click.option('--cli-profile', default='docs-ci', help='Databricks CLI profile name.')
+@click.option('--workspace-tmp-dir', default='/tmp/glow-docs-ci', help='Temp workspace dir for import/export.')
+def main(html, cli_profile, workspace_tmp_dir):
+    assert os.path.commonpath([NOTEBOOK_DIR, html]) == NOTEBOOK_DIR, \
+        "HTML notebook must be under {} but got {}.".format(NOTEBOOK_DIR, html)
+    rel_path = os.path.splitext(os.path.relpath(html, NOTEBOOK_DIR))[0]
+
+    if not os.path.exists(html):  # html notebook was deleted
+        print("{} does not exist. Deleting the companion source file...".format(html))
+        for ext in SOURCE_EXTS:
+            source_path = os.path.join(SOURCE_DIR, rel_path + "." + ext)
+            if os.path.exists(source_path):
+                os.remove(source_path)
+                print("\tDeleted {}.".format(source_path))
+        return
+
+    print("Generating source file for {} under {} ...".format(html, SOURCE_DIR))
+
+    def run_cli_workspace_cmd(args):
```

Comment:
Why is this a nested definition?

Reply:
This is just so we can leverage `cli_profile` from the main args, but I can split this out if it's clearer that way.

Reply:
I think that would be a bit cleaner.